internal/core/items/text.rs

```diff
@@ -479,6 +479,11 @@ impl Item for TextInput {
                 {
                     return KeyEventResult::EventIgnored;
                 }
+                if matches!(self.input_type(), InputType::Number)
```

Comment:
I put this check after `text.insert_str(insert_pos, &event.text);` because I need the updated `text` to verify if it is valid decimal.

For example, user inputs `1, 2, 3`, then move cursor to the head, input `-`, then move cursor to the head again & input `-`, this can be detected and disallowed because the parsing for `--123` will fail.

If the check is placed after the shortcuts but before the `text.insert_str(...)`, although I have the info of `self.text()` and `event.text` but I don't have the info of insert position, what I can only do is assume `event.text` will append to `self.text()`.

Reply:
@ogoffart Need your input here for my concern :)

Reply:
I think it should go after the shortcut handling, together with the decimal handling.